app/models/effective/page.rb

```diff
@@ -76,6 +110,14 @@ class Page < ActiveRecord::Base
       published.where(menu: false).or(published.where(menu: true).where.not(id: menu_root_with_children))
     }
 
+    # Maybe change paginate to be an isolated function instead of a scope, so we can use it with PgSearch::Document when doing a whole app search?
```

Comment:
What would be involved in doing this?

You're talking about like a class level function, or an instance method?

Reply:
An "isolated function" to me, means concurrency safe. I don't expect any race conditions here, we don't have anything multi threaded